Reduce dependencies on RecoEgamma and requirements for RecoEgamma

guitargeek opened the pull request:
#### PR description:

To make cmssw a bit more modular, I looked at the RecoEgamma packages and tried to remove all unused requirements. At the same time, I looked at the spots where other parts of cmssw depend on RecoEgamma and tried to remove these dependencies where possible. The ElectronIdentification and PhotonIdentification packages got special attention, and all sources in these libraries which are only used for plugins were moved into the plugins directory to make this more explicit.

#### PR validation:

CMSSW compiles and local matrix tests pass.

#### if this PR is a backport please specify the original PR:

No backport intended.

guitargeek committed:
reduce dependencies of RecoEgamma

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28684/13255

- This PR adds an extra 208KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/Trigger/plugins/LepHTMonitor.cc  modified in PR(s): #28552, #28627, #28622
  - File  Validation/RecoEgamma/plugins/PhotonValidator.h  modified in PR(s): #28552, #28622

cmsbuild commented:
A new Pull Request was created by  @guitargeek (Jonas Rembser) for master.

It involves the following packages:

CommonTools/RecoUtils
Configuration/Skimming
DQM/Physics
DQM/TrackingMonitorSource
DQMOffline/EGamma
DQMOffline/Ecal
DQMOffline/Trigger
ElectroWeakAnalysis/WENu
HLTrigger/Egamma
L1Trigger/L1TNtuples
RecoBTag/SoftLepton
RecoEcal/EgammaClusterAlgos
RecoEcal/EgammaClusterProducers
RecoEgamma/EgammaElectronAlgos
RecoEgamma/EgammaElectronProducers
RecoEgamma/EgammaIsolationAlgos
RecoEgamma/EgammaPhotonAlgos
RecoEgamma/EgammaPhotonProducers
RecoEgamma/ElectronIdentification
RecoEgamma/Examples
RecoEgamma/PhotonIdentification
RecoParticleFlow/PFProducer
RecoParticleFlow/PFTracking
TrackingTools/TrajectoryState
Validation/RecoEgamma


@perrotta, @kmaeshima, @benkrikler, @cmsbuild, @chayanit, @zhenhu, @fwyzard, @schneiml, @andrius-k, @pgunnell, @rekovic, @jfernan2, @fioriNTU, @Martin-Grunewald, @slava77, @santocch can you please review it and eventually sign? Thanks.
@rappoccio, @felicepantaleo, @jainshilpi, @hatakeyamak, @emilbols, @argiro, @HeinerTholen, @fioriNTU, @thomreis, @abbiendi, @varuns23, @seemasharmafnal, @mmarionncern, @smoortga, @hdelanno, @JyothsnaKomaragiri, @makortel, @threus, @acaudron, @jhgoh, @lgray, @missirol, @HuguesBrun, @cericeci, @ferencek, @trocino, @rociovilar, @Sam-Harper, @cbernet, @GiacomoSguazzoni, @rovere, @VinInn, @bellan, @jandrea, @idebruyn, @ebrondol, @mtosi, @dgulhan, @kreczko, @Martin-Grunewald, @battibass, @Fedespring, @calderona, @mverzett, @sobhatta, @folguera, @afiqaize, @gpetruc, @andrzejnovak, @pvmulder, @bachtis this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos, @silviodonato you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 1fdd8e2bee77e1512b420edb43deb599ce3a7a66
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ef0eba/4105/summary.html
CMSSW: CMSSW_11_1_X_2019-12-26-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ef0eba/4105/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2815765
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2815417
 - DQMHistoTests: Total skipped: 346
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

Martin-Grunewald commented:
+1

perrotta reviewed (commented):
Nice cleaning.
Maybe you could further adjust the header guards in those header files that you moved somewhere else, where needed.

guitargeek committed:
FTSFromVertexToPointFactory to free function

cmsbuild commented:
The code-checks are being triggered in jenkins.

guitargeek commented:
Hi @perrotta, thanks for you comments! I implemented them both in the last commit and additionally renamed the now moved `FTSFromVertexToPointFactory.h` because it was not really a factory. just a free function.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28684/13279

- This PR adds an extra 44KB to repository

- Found files with invalid states:
  - TrackingTools/TrajectoryState/interface/FTSFromVertexToPointFactory.h:
    - Added: 1fdd8e2bee77e1512b420edb43deb599ce3a7a66
    - Deleted: bd2bed609d12aeba248fc6735997ff00e93111fa
  - TrackingTools/TrajectoryState/src/FTSFromVertexToPointFactory.cc:
    - Added: 1fdd8e2bee77e1512b420edb43deb599ce3a7a66
    - Deleted: bd2bed609d12aeba248fc6735997ff00e93111fa


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/Trigger/plugins/LepHTMonitor.cc  modified in PR(s): #28552, #28622, #28627
  - File  Validation/RecoEgamma/plugins/PhotonValidator.h  modified in PR(s): #28552, #28622
  - File  Validation/RecoEgamma/plugins/TkConvValidator.cc  modified in PR(s): #28687

cmsbuild commented:
Pull request #28684 was updated. @perrotta, @kmaeshima, @benkrikler, @cmsbuild, @chayanit, @zhenhu, @fwyzard, @schneiml, @andrius-k, @pgunnell, @rekovic, @jfernan2, @fioriNTU, @Martin-Grunewald, @slava77, @santocch can you please check and sign again.

perrotta commented on TrackingTools/TrajectoryState/interface/ftsFromVertexToPoint.h:
```diff
@@ -1,9 +1,7 @@
-// Class:      FTSFromVertexToPointFactory
 //
 /**
  *  Generates a FreeTrajectoryState from a given measured point, vertex
```
The description should be updated to reflect your last change.
Also add your name as "Modified by" after L16

guitargeek replied:
Which part exactly should I update, you want me to put this package tag back in with the new location? Or is it fine leaving it out and I should just update the description text with the example? Thanks

perrotta replied:
I was thinking just at the example, when it says
```
 *  FTSFromVertexToPointFactory myFTS; myFTS(xmeas, xvert, momentum, charge);
 *  gives a FreeTrajectoryState of a track which comes from xvert to xmeas. 
```
Everything else seems still up to date to me

guitargeek committed:
slim down interface of EgammaHadTower

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28684/13280

- This PR adds an extra 296KB to repository

- Found files with invalid states:
  - TrackingTools/TrajectoryState/interface/FTSFromVertexToPointFactory.h:
    - Added: 7095fb0bab4ac0faf96234237db0f36dae7ab663
    - Deleted: 8a04adadbada51096ea9aa316c9866f1724bd539
  - TrackingTools/TrajectoryState/src/FTSFromVertexToPointFactory.cc:
    - Added: 7095fb0bab4ac0faf96234237db0f36dae7ab663
    - Deleted: 8a04adadbada51096ea9aa316c9866f1724bd539


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/Trigger/plugins/LepHTMonitor.cc  modified in PR(s): #28552, #28622, #28627
  - File  Validation/RecoEgamma/plugins/PhotonValidator.h  modified in PR(s): #28552, #28622
  - File  Validation/RecoEgamma/plugins/TkConvValidator.cc  modified in PR(s): #28687

cmsbuild commented:
Pull request #28684 was updated. @perrotta, @kmaeshima, @benkrikler, @cmsbuild, @chayanit, @schneiml, @tlampen, @christopheralanwest, @andrius-k, @tocheng, @pgunnell, @rekovic, @franzoni, @jfernan2, @fioriNTU, @Martin-Grunewald, @slava77, @fwyzard, @pohsun, @santocch, @zhenhu can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 7894aaa45bf848ee3177283df562565c8a5cd9fd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ef0eba/4126/summary.html
CMSSW: CMSSW_11_1_X_2020-01-02-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ef0eba/4126/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2815513
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2815166
 - DQMHistoTests: Total skipped: 346
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

jfernan2 commented:
+dqm

perrotta commented:
+1
- Code cleaning applied as outlined in the PR description
- As such, the changes are rather technical: no effect is visible on jenkins tests outputs, nor any difference in timing performance was measured in the 10224 (TTbar with PU) workflow

santocch commented:
+1

Martin-Grunewald commented:
+1

tlampen commented:
+1

silviodonato commented:
@benkrikler  @rekovic @chayanit  @pgunnell  @zhenhu please review/sign

cmsbuild commented:
The code-checks are being triggered in jenkins.

guitargeek commented:
Hi, unfortunately I had to rebase to fix a conflict in `DQMOffline/Trigger/plugins/LepHTMonitor.cc`.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28684/13334

- This PR adds an extra 472KB to repository

- Found files with invalid states:
  - TrackingTools/TrajectoryState/interface/FTSFromVertexToPointFactory.h:
    - Added: 9766c72b5616e49b56c80092018d025c9a649c73
    - Deleted: 67121c6cf0ed1a58f509e4c8c9a3a6f3c3ccba47
  - TrackingTools/TrajectoryState/src/FTSFromVertexToPointFactory.cc:
    - Added: 9766c72b5616e49b56c80092018d025c9a649c73
    - Deleted: 67121c6cf0ed1a58f509e4c8c9a3a6f3c3ccba47


- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Validation/RecoEgamma/plugins/PhotonValidator.h  modified in PR(s): #28622

cmsbuild commented:
Pull request #28684 was updated. @perrotta, @kmaeshima, @benkrikler, @chayanit, @schneiml, @tlampen, @christopheralanwest, @andrius-k, @tocheng, @cmsbuild, @rekovic, @franzoni, @jfernan2, @fioriNTU, @Martin-Grunewald, @slava77, @fwyzard, @pohsun, @santocch, @pgunnell, @zhenhu can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: b4ebcd1821622801f32808481157d6eff3551de3
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ef0eba/4193/summary.html
CMSSW: CMSSW_11_1_X_2020-01-09-2300
SCRAM_ARCH: slc7_amd64_gcc820

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #28638 @mteroerd: APE estimation tool: Enable submission of skims to condor, rearrange skimming tool, add phase 0 geometry support for plotting tools.
 - #28429 @guitargeek: Phase out old pre-GED GsfElectron sequence
 - #28658 @vberta: quark and gluons Py8EGun

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ef0eba/4193/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ef0eba/4193/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ef0eba/4193/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2696324
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2695977
 - DQMHistoTests: Total skipped: 346
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

jfernan2 commented:
+1
No dqm changes

perrotta commented:
+1
- Only rebased since last reco signature in https://github.com/cms-sw/cmssw/pull/28684#issuecomment-570802837

tocheng commented:
+1

chayanit commented:
+1

santocch commented:
+1

perrotta commented:
Just to avoid having to rebase once more:
- @benkrikler,@rekovic could you please have a look and sign, if you think so, for L1?
- @Martin-Grunewald you already signed for HLT before the rebase, I think you can redo it now

Martin-Grunewald commented:
+1

silviodonato commented:
+1

silviodonato commented:
merge
@rekovic this PR changes only one line of L1
